prime/src/components/graph/PnLGraph.tsx

```diff
@@ -252,23 +259,74 @@ export default function PnLGraph(props: PnLGraphProps) {
     setBorrowInterestInputValue,
     setSwapFeesInputValue,
   } = props;
+  const { activeChain } = useContext(ChainContext);
   const [data, setData] = useState<Array<PnLEntry>>([]);
   const [localInTermsOfToken0, setLocalInTermsOfToken0] = useState<boolean>(inTermsOfToken0);
+  const [priceAtLastUpdate, setPriceAtLastUpdate] = useState<number | null>(null);
 
-  let price = sqrtRatioToPrice(
+  useEffect(() => {
+    let mounted = true;
+    async function fetch() {
+      let etherscanResult: AxiosResponse<any, any> | null = null;
+      try {
+        etherscanResult = await makeEtherscanRequest(
+          0,
+          ALOE_II_FRONTEND_MANAGER_ADDRESS,
+          [TOPIC0_MODIFY_EVENT, `0x000000000000000000000000${marginAccount.address.slice(2)}`],
+          true,
+          activeChain
+        );
+      } catch (e) {
+        console.error(e);
+      }
+      if (etherscanResult == null || !Array.isArray(etherscanResult.data.result)) return [];
+      if (etherscanResult.data.result.length === 0) return [];
+      const mostRecentEvent = etherscanResult.data.result[0];
+      const mostRecentEventData: BigNumber = ethers.utils.defaultAbiCoder.decode(['uint256'], mostRecentEvent.data)[0];
+      const mostRecentPrice = tickToPrice(
+        mostRecentEventData.toNumber(),
+        marginAccount.token0.decimals,
+        marginAccount.token1.decimals
+      );
+      if (mounted) {
+        setPriceAtLastUpdate(mostRecentPrice);
+      }
+    }
+    fetch();
+    return () => {
+      mounted = false;
+    };
+  }, [
+    activeChain,
+    activeChain.id,
+    marginAccount.address,
+    marginAccount.token0.decimals,
+    marginAccount.token1.decimals,
+  ]);
+
+  let currentPrice = sqrtRatioToPrice(
     marginAccount.sqrtPriceX96,
     marginAccount.token0.decimals,
     marginAccount.token1.decimals
   );
-  if (inTermsOfToken0) price = 1 / price;
-  const priceA = price / PLOT_X_SCALE;
-  const priceB = price * PLOT_X_SCALE;
+  let previousPrice = priceAtLastUpdate || currentPrice;
+
+  // If we're showing hypothetical, we want to use the current price as the price at the last update
+  if (isShowingHypothetical) previousPrice = currentPrice;
+
+  if (inTermsOfToken0) {
+    currentPrice = 1 / currentPrice;
+    previousPrice = 1 / previousPrice;
+  }
+  const priceA = previousPrice / PLOT_X_SCALE;
+  const priceB = previousPrice * PLOT_X_SCALE;
```

Comment:
We might want to adjust these and/or `PLOT_X_SCALE` in order to force both currentPrice and previousPrice to be visible